bob-backend/src/core.rs

```diff
@@ -149,7 +216,12 @@ impl Backend {
                 Arc::new(pearl)
             }
         };
-        Self { inner, mapper }
+
+        let error_logger = 
+            Arc::new(Mutex::new(BackendErrorLogger::new(config.error_log_interval())));
+        BackendErrorLogger::spawn_task(error_logger.clone());
```

Comment:
Too much complexity for simple logging. This task can be done without additional thread by measuring time intervals. Example on C#: https://github.com/qoollo/dotNet-turbo/blob/master/src/Qoollo.Turbo/Common/EventTimingTracker.cs

Reply:
For fast time measurement use this crate: https://docs.rs/coarsetime/latest/coarsetime/